src/main/java/io/github/zap/commons/keyvaluegetters/ValueDeserializer.java

```diff
@@ -0,0 +1,7 @@
+package io.github.zap.commons.keyvaluegetters;
+
+import java.lang.reflect.ParameterizedType;
+
+public interface ValueDeserializer<T> {
+    OperationResult<T> deserialize(String value, ParameterizedType pt, DeserializerEngine deserializers);
```

Comment:
I suspect that OperationResult<T> will  never normally be null so this should be @NotNull

Reply:
yep. I just realized I forget to add a lot of @Nullable and @NotNull